internal/controller/controllers/infraenv_controller.go

```diff
@@ -183,10 +184,46 @@ func (r *InfraEnvReconciler) updateInfraEnv(ctx context.Context, log logrus.Fiel
 
 	updateParams.InfraEnvUpdateParams.ImageType = r.Config.ImageType
 
+	existingKargs, err := kubeKernelArgs(internalInfraEnv)
+	if err != nil {
+		return nil, err
+	}
+	if !funk.Equal(infraEnv.Spec.KernelArguments, existingKargs) {
+		updateParams.InfraEnvUpdateParams.KernelArguments = internalKernelArgs(infraEnv.Spec.KernelArguments)
+	}
+
 	// UpdateInfraEnvInternal will generate an ISO only if there it was not generated before,
 	return r.Installer.UpdateInfraEnvInternal(ctx, updateParams, nil)
 }
 
+func kubeKernelArgs(internalInfraEnv *common.InfraEnv) ([]aiv1beta1.KernelArgument, error) {
+	var args models.KernelArguments
+	if internalInfraEnv.KernelArguments != nil {
+		if err := json.Unmarshal([]byte(swag.StringValue(internalInfraEnv.KernelArguments)), &args); err != nil {
+			return nil, errors.Wrapf(err, "failed to unmarshal discovery kernel arguments of infra-env %s", internalInfraEnv.ID.String())
+		}
+	}
+	var ret []aiv1beta1.KernelArgument
```

Comment:
just an observation: generally when transforming a slice of X to a slice of Y, we know the length, so it's helpful to `make()` the slice the correct length up-front and assign values as you go.

```
ret := make([]aiv1beta1.KernelArgument, len(args))
for i := range args {
  ret[i] = ...
}
```

This helps avoid re-sizing and fragmentation of the slice's underlying data, though for cases like this it probably doesn't make much difference.

Reply:
Right, usually the slice size for kernel arguments is small.